src/profiles.js

```diff
@@ -0,0 +1,77 @@
+import tv4 from 'tv4'
+import jsonlint from 'json-lint'
+import _ from 'lodash'
+
+import Registry from './registry'
+
+class Profiles {
+
+  constructor(remote = false) {
```

Comment:
@roll if I understand you correctly, I think that isn't possible. We can run the promises in the constructor but we can not return promise on the class initialization, so the consumer would not know if the methods he will try to access will return the expected value or not (depending of the promises ran in the constructor).